Fix nit which is inefficient sequence to deleteMarkedNodes

Ahmad-S792 opened the pull request:
#### 6eceb6d922f3d1ba8cd0da21fc457249938cef87
<pre>
Fix nit which is inefficient sequence to deleteMarkedNodes

Fix nit which is inefficient sequence to deleteMarkedNodes
<a href="https://bugs.webkit.org/show_bug.cgi?id=250524">https://bugs.webkit.org/show_bug.cgi?id=250524</a>

Reviewed by Eric Carlson.

Merge - <a href="https://src.chromium.org/viewvc/blink?view=rev&amp">https://src.chromium.org/viewvc/blink?view=rev&amp</a>;revision=172279

This patch is just to move &quot;deleteMarkedNodes&quot; just after &apos;m_nodesToDelete&apos; to prevent inefficient do while loop.

* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
(BaseAudioContext::clear): Move &apos;deleteMarkedNodes&apos; down

Canonical link: <a href="https://commits.webkit.org/258920@main">https://commits.webkit.org/258920@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3da007a90b15215bd1c5301a0ae279ddac5bb42

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103129 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12255 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36148 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112377 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172575 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107083 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13275 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3157 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95367 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/110601 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108903 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10195 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93394 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37831 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92026 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24938 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79565 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5665 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26344 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5832 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2788 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11825 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45847 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6131 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7583 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
Thank you Ahmad!

r=me

Ahmad-S792 committed:
Fix nit which is inefficient sequence to deleteMarkedNodes

Fix nit which is inefficient sequence to deleteMarkedNodes
https://bugs.webkit.org/show_bug.cgi?id=250524

Reviewed by Eric Carlson.

Merge - https://src.chromium.org/viewvc/blink?view=rev&revision=172279

This patch is just to move "deleteMarkedNodes" just after 'm_nodesToDelete' to prevent inefficient do while loop.

* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
(BaseAudioContext::clear): Move 'deleteMarkedNodes' down

Canonical link: https://commits.webkit.org/258920@main
